pkg/controller/deployment/deployment_controller.go

```diff
@@ -413,9 +430,56 @@ func (dc *DeploymentController) enqueueDeployment(deployment *extensions.Deploym
 	// periodically relist all deployments there will still be some ReplicaSet instability. One
 	//  way to handle this is by querying the store for all deployments that this deployment
 	// overlaps, as well as all deployments that overlap this deployments, and sorting them.
+
+	// Relist all deployment for overlaps, and avoid syncing the newer overlapping ones (only sync the oldest one)
+	deployments, err := dc.dStore.List()
```

Comment:
> This is too expensive to happen on every enqueue. @deads2k got any ideas for fighting overlapping selectors here? How about indexing deployments with a StringSelectorIndex and skip enqueueing every time dc.dStore.ByIndex(StringSelectorIndex, deployment.Spec.Selector) returns deployments of the same namespace?

I agree that this is too expensive, but only finding ones with the same index doesn't do the same thing since you can overlap without being the same.

Limiting to deployments in the same namespace and doing the check when work is going to be performed (on dequeue so that duplicate enqueues don't pay the tax multiple times) is probably enough.